lib/ProductOpener/DataQualityFood.pm

```diff
@@ -1302,6 +1302,27 @@ sub check_nutrition_data ($product_ref) {
 
 		}
 
+
+# sum of nutriments that compose fiber can not be greater than fiber value
+if (
+    (defined $product_ref->{nutriments}{fiber_100g})
+    and (
+        (
+            (
+                (defined $product_ref->{nutriments}{soluble_fiber_100g}) ? $product_ref->{nutriments}{soluble_fiber_100g}
```

Comment:
```suggestion
                (defined $product_ref->{nutriments}{"soluble-fiber_100g"}) ? $product_ref->{nutriments}{soluble_fiber_100g}
```

It's difficult to spot, but if you look at the JSON you got for your product, the field name is soluble-fiber_100g (with a dash between soluble and fiber) and not soluble_fiber_100g

Same change is needed for insoluble fiber

Reply:
@stephanegigandet Thanks for the review, I have corrected the error. And am getting the correct error message when soluble fiber is plus is insoluble fiber is greater than the fiber value.
<img width="1680" alt="Screenshot 2024-03-14 at 18 32 08" src="https://github.com/openfoodfacts/openfoodfacts-server/assets/85356158/58b02e44-18d1-4455-9d28-3c382d26b4e6">